File: B_Parity_Sort.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
   int t;                  cin>>t;
   while(t--){
      int n;               cin>>n;
      vector<int> a(n);
      for(int i=0;i<n;i++){
         cin>>a[i];
      }

      vector<bool> isOdd(n,false);
      vector<int> even,odd;
      for(int i=0;i<n;i++){
         if(a[i]%2==1){
            isOdd[i]=true;
            odd.push_back(a[i]);
         }
         else{
            even.push_back(a[i]);
         }
      }

      sort(even.begin(),even.end(),greater<int>());
      sort(odd.begin(),odd.end(),greater<int>());

      for(int i=0;i<n;i++){
         if(isOdd[i]){
            a[i]=odd.back();
            odd.pop_back();
         }
         else{
            a[i]=even.back();
            even.pop_back();
         }
      }


      if(is_sorted(a.begin(),a.end())){
         cout<<"YES"<<endl;
      }
      else{
         cout<<"NO"<<endl;
      }


   }
   return 0;
}

File: B_Parity_Sort.py
def main():
    t = int(input())
    while t > 0:
        t -= 1
        n = int(input())
        a = list(map(int, input().split()))

        isOdd = [False] * n
        odd, even = [], []

        for i in range(n):
            if a[i] % 2 == 1:
                isOdd[i] = True
                odd.append(a[i])
            else:
                even.append(a[i])

        even.sort(reverse=True)
        odd.sort(reverse=True)

        for i in range(n):
            if isOdd[i]:
                a[i] = odd.pop()
            else:
                a[i] = even.pop()

        if a == sorted(a):
            print("YES")
        else:
            print("NO")

if __name__ == "__main__":
    main()